src/test.tsx

```diff
@@ -0,0 +1,216 @@
+import { useState, useEffect } from 'react'
+
+interface User {
+  id: number
+  name: string
+  status: any
+}
+
+const users: User[] = [
+  { id: 1, name: 'João', status: 'active' },
+  { id: 2, name: 'Maria', status: 'inactive' },
+  { id: 3, name: 'Pedro', status: 'active' },
+]
+
+function App() {
+  const [selectedUserId, setSelectedUserId] = useState<number | null>(null)
+  const [count, setCount] = useState(0)
+  const [isVisible, setIsVisible] = useState(true)
+  const [totalActiveUsers, setTotalActiveUsers] = useState(0)
```

Comment:
Não precisamos de estados no react para recalcular informações, pois o próprio render do react já vai fazer com que voce consiga processar isso sem que vc precise de um estado.

Dê uma olhada nesse artigo que ele explica sobre isso: https://react.dev/learn/you-might-not-need-an-effect

```tsx

// remove o state e o useEffec e mantenha apenas isso no corpo do component
const totalActiveUsers = users.filter(user => user.status === 'active').length
```